girder/api/v1/notification.py

```diff
@@ -121,8 +121,16 @@ def streamGen():
     def listNotifications(self, params):
         user, token = self.getCurrentUser(returnToken=True)
         since = params.get('since')
-        setResponseHeader('Date', str(time.time()))
 
         if since is not None:
             since = datetime.utcfromtimestamp(since)
-        return list(NotificationModel().get(user, since, token=token))
+        notifications = list(NotificationModel().get(user, since, token=token))
+        if not notifications:
+            timestamp = time.time()
```

Comment:
This `time.time()` should be computed prior to the query so we don't miss any notifications.